sklearn/cross_decomposition/tests/test_pls.py

```diff
@@ -383,52 +383,65 @@ def test_copy(Est):
 
 
 @pytest.mark.parametrize('Est', (CCA, PLSCanonical, PLSRegression, PLSSVD))
-def test_scale_and_stability(Est):
+def test_scale_and_stability_non_regression_7819(Est):
```

Comment:
Instead of having different tests, can we just parametrize the datasets? This one is lacking a description now

Reply:
It might be good to also parametrize the test to check the svd solver and not just the power method

Reply:
I find parametrizing datasets no so readable. I prefer a little bit of redundancy but readability.

Reply:
OK, but I guarantee that someone in the future will come and factorize these ;)

(@thomasjpfan  please make sure all tests are equally documented)

Reply:
Actually we could just name the datasets and keep all the dataset generation code in the body of the test and use and if / elif branch on the dataset name in the test body, indeed. As you wish.

What I don't like is putting the datasets generation in private helpers and passing the dataset generators it as a parametrize argument to the test.

Reply:
You mean parametrizing names? I'd prefer that yeah.
Just a question: you don't like parametrizing the actual datasets + using ids?

Reply:
Because you have an indirection to read the data generating process when you are reading the test and trying to make sense on their relationship between the 2.